modules/effects/src/effect_sources.ts

```diff
@@ -29,22 +21,42 @@ export class EffectSources extends Subject<any> {
    */
   toActions(): Observable<Action> {
     return this.pipe(
-      groupBy(getSourceForInstance),
-      mergeMap(source$ =>
-        source$.pipe(
-          exhaustMap(resolveEffectSource),
-          map(output => {
-            verifyOutput(output, this.errorHandler);
+      scan(
```

Comment:
I might be overlooking something but won't the following accomplish the same thing?

```ts
groupBy(source => source),
```

Reply:
You are completely right, thanks for noticing! I will update the PR.